internal/controllers/configuration/zz_generated_controllers.go

```diff
@@ -420,7 +424,7 @@ func (r *NetV1IngressReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	// if status updates are enabled report the status for the object
 	if r.DataplaneClient.AreKubernetesObjectReportsEnabled() {
 		log.V(util.DebugLevel).Info("determining whether data-plane configuration has succeeded", "namespace", req.Namespace, "name", req.Name)
-		if !r.DataplaneClient.KubernetesObjectIsConfigured(obj) {
+		if !(r.DataplaneClient.KubernetesObjectIsConfigured(obj) == k8sobj.ConfiguredStatusSucceeded) {
```

Comment:
The controllers that've been using the existing mechanism for successful status reporting can be adapted without breaking changes, and without handling error cases. That can be incrementally done in separate steps